frontend/src/components/MultiSelection.tsx

```diff
@@ -9,46 +23,179 @@ type MultiSelectionProps = {
   ariaLabel: string;
 };
 
-export const MultiSelection: React.FC<MultiSelectionProps> = ({ value, setValue, ariaLabel }) => {
-  const [showMenu, setShowMenu] = React.useState(false);
+export const MultiSelection: React.FC<MultiSelectionProps> = ({ value, setValue }) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [inputValue, setInputValue] = React.useState<string>('');
+  const [focusedItemIndex, setFocusedItemIndex] = React.useState<number | null>(null);
+  const [activeItem, setActiveItem] = React.useState<string | null>(null);
+  const textInputRef = React.useRef<HTMLInputElement>();
+  const selected = React.useMemo(() => value.filter((v) => v.enabled), [value]);
+  const selectOptions = React.useMemo(
+    () =>
+      value.filter((v) => !inputValue || v.name.toLowerCase().includes(inputValue.toLowerCase())),
+    [inputValue, value],
+  );
+
+  React.useEffect(() => {
+    if (inputValue) {
+      setIsOpen(true);
+    }
+    setFocusedItemIndex(null);
+    setActiveItem(null);
+  }, [inputValue]);
+
+  const handleMenuArrowKeys = (key: string) => {
+    let indexToFocus;
+    if (isOpen) {
```

Comment:
Since this is back again: if it is not open, it should be opened and just return.